packages/docregistry/src/context.ts

```diff
@@ -203,7 +202,6 @@ export class Context<
     this._isDisposed = true;
     this.sessionContext.dispose();
     this._model.dispose();
-    this._provider.dispose();
     this._model.sharedModel.dispose();
```

Comment:
```suggestion
    // Ensure we dispose the `sharedModel` as it may have been generated in the context
    // through the shared model factory.
    this._model.sharedModel.dispose();
```